### PR TITLE
[SCOUT-250] unassigned tab

### PR DESCRIPTION
**Pull request by jakubfiglak**
⚠️ REQUIRES https://github.com/scoutmaker/scoutmaker-backend-v2-nest/pull/140

[SCOUT-250](https://playmakerpro.atlassian.net/browse/SCOUT-250)

### Task Description

- Display unassigned notes table on single match page (http://localhost:3000/matches/{matchId})

https://user-images.githubusercontent.com/55458549/203153535-74b03768-a3ce-4fa0-97d6-c6cab47537d4.mov

And a bunch of refactoring ideas/proposals:

- render tabs by looping through a predefined array of data in order to stay DRY inside of jsx
- use `useTabs` hook to manage tabs
- ⚠️ EXPERIMENT: I felt like this whole page file has become hard to read and was doing a little too much by itself so I decided to move all the data fetching logic to a custom hook (this hook is not meant to be reused anywhere else, I'm extracting it to improve readability). @dejwi please let me know what you think about this.

### CODE REVIEW

File: src/pages/matches/[id].tsx [hunk 1]
```diff
@@ -151,74 +96,57 @@ const MatchPage = ({ data, errorMessage, errorStatus }: TSsrRole<MatchDto>) => {
...
-            <Tab label={`${t('NOTES')} (${awayTeamNotes?.totalDocs || 0})`} />
-            <Tab
-              label={`${t('REPORTS')} (${awayTeamReports?.totalDocs || 0})`}
-            />
+            {tabs.map(({ name, type, docCount }) => (
+              <Tab
+                key={name}
+                label={`${
+                  type === 'note' ? t('NOTES') : t('REPORTS')
+                } (${docCount})`}
```
Comment (dejwi) [inline src/pages/matches/[id].tsx:111]: can't types be `NOTES` and `REPORTS` to do just that?
```suggestion
                label={`${t(type)} (${docCount})`}
```

**Review by dejwi — CHANGES_REQUESTED** [review]
Nicee smaller code is always better, one small question about type names below and what about mobile sizing?
There's too many tabs and they don't fit there with this new section, maybe we can combine 2 teams sections into one where team name is just some slick select input? idk
image

File: src/pages/matches/[id].tsx (hunk 1 above)
Comment (dejwi) [inline src/pages/matches/[id].tsx:111]: not some important stuff but can't types be `NOTES` and `REPORTS` to do just that

```suggestion
                label={`${t(type)} (${docCount})`}
```

Comment (jakubfiglak) [reply src/pages/matches/[id].tsx:111]: Respectfully disagree - image if someone comes to do some changes here later and for some reason decides to change the `type` name on our object - how is that person supposed to know that it's gonna break translations? Or the other way around - what if someone wants to change translation key for the heading? Is that person supposed to change the `type`? Then the condition breaks.

Long story short - in this case `type` and i18n key are similar only by coincidence and they shouldn't be dependent on each other

**Comment by dejwi** [comment]
> What is the issue on this screenshot? Tabs do scroll horizontally don't they? Like with the red arrow on the below screenshot.
> 
> The only thing that bothers me here is the section I marked with the blue rectangle, I can try and tweak font size and padding there
> 
> Screenshot 2022-11-23 at 09 01 02

I was not able to scroll tabs at least on my pc, if you’ll work on this task before I’ll have free time please double check that

**Comment by jakubfiglak** [comment]
> > What is the issue on this screenshot? Tabs do scroll horizontally don't they? Like with the red arrow on the below screenshot.
> > The only thing that bothers me here is the section I marked with the blue rectangle, I can try and tweak font size and padding there
> > Screenshot 2022-11-23 at 09 01 02
> 
> I was not able to scroll tabs at least on my pc, if you’ll work on this task before I’ll have free time please double check that (if this is the case maybe a overflow-x: scroll would fix that or whatever it was)

You're right, it doesn't work well on smaller devices. I need to think of solution to this, good catch.

**Comment by jakubfiglak** [comment]
How about now? Slightly different approach

### DESKTOP

Uploading Screen Recording 2022-11-23 at 19.18.50.mov…

### MOBILE

https://user-images.githubusercontent.com/55458549/203620500-eba9ca08-eef7-43af-ba65-bd97fb048d9c.mov

**Comment by jakubfiglak** [comment]
@dejwi can I get another review please?

**Review by dejwi — APPROVED** [review]
Works a lot better:)
